xrpl/models/ledger_objects/ledger_object.py

```diff
@@ -0,0 +1,137 @@
+"""A model for Ledger Objects"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Type, TypeVar
+
+from xrpl.models.base_model import BaseModel
+from xrpl.models.exceptions import XRPLModelException
+from xrpl.models.ledger_objects.ledger_entry_type import LedgerEntryType
+
+# TODO: REMOVE if optional is needed
+# from xrpl.models.required import REQUIRED
+from xrpl.models.utils import require_kwargs_on_init
+
+L = TypeVar("L", bound="LedgerObject")
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class LedgerObject(BaseModel):
+    """The base model for a Ledger Object."""
+
+    # TODO: Try without optional
+    ledger_entry_type: Optional[LedgerEntryType] = None
+    index: Optional[str] = None
```

Comment:
These shouldn't need to be optional.

Reply:
`LedgerEntryType` should be required but `index` is optional according to docs - https://xrpl.org/ledger-entry-common-fields.html